example/burrow_integration/setupAccount.sol

```diff
@@ -0,0 +1,143 @@
+        (bool success, bytes memory ret) = address(serviceContractAddress).delegatecall(payload);
+        require(success, "Error calling service contract function");
+
+        emit SubtractedAmount(asset, amount);
+        result = ret;
+    }
+
+    //Queries balance of an iroha account
+    function queryBalance(string memory _account, string memory _asset) public returns (bytes memory result) {
+        bytes memory payload = abi.encodeWithSignature(
+            "getAssetBalance(string,string)",
+            _account,
+            _asset);
+        (bool success, bytes memory ret) = address(serviceContractAddress).delegatecall(payload);
+        require(success,"Error calling service contract function ");
+        result = ret;
+    }
+
+     // Sets the account detail
+    function setAccountDetail(string memory name, string memory key, string memory value) public returns (bytes memory result) {
+        bytes memory payload = abi.encodeWithSignature(
+            "setAccountDetail(string,string,string)",
+            name,
+            key,
+            value);
+        (bool success, bytes memory ret) = address(serviceContractAddress).delegatecall(payload);
+        require(success, "Error calling service contract function");
+        emit Updated(name, key, value);
+        result = ret;
+    }
+
+    // Gets account details
+    function getAccountDetail() public returns (bytes memory result) {
+         bytes memory payload = abi.encodeWithSignature(
+            "getAccountDetail()");
+        (bool success, bytes memory ret) = address(serviceContractAddress).delegatecall(payload);
+        require(success, "Error calling service contract function");
+        result = ret;
+    }
+
+    //Transfers asset from one iroha account to another
+    function transferAsset(string memory src, string memory dst, string memory asset, string memory description, string memory amount) public returns (bytes memory result) {
+        bytes memory payload = abi.encodeWithSignature(
+            "transferAsset(string,string,string,string,string)",
+            src,
+            dst,
+            asset,
+            description,
+            amount);
+        (bool success, bytes memory ret) = address(serviceContractAddress).delegatecall(payload);
+        require(success, "Error calling service contract function");
+
+        emit Transferred(src, dst, amount);
+        result = ret;
+    }
+
+    function createAndSetupAccount(string memory adminAccountId, string memory userAccountName, string memory userAccountId, string memory assetId, string memory domain, string memory description, string memory amount, string memory key, string memory value, string memory publicKey) public returns (bytes memory result) {
+        result = createAccount(userAccountName, domain, publicKey);
+        result = transferAsset(adminAccountId, userAccountId, assetId, description, amount);
+        result = setAccountDetail(userAccountId, key, value);
```

Comment:
I suggest to add checking: if result is error then don't call another function.

Reply:
@baziorek @Ayush-Jalan This PR is rather big so I would suggest creating a new one for documentation and just referencing this one in the description. Please note that we usually start PR names for documentation with something like "Docs:...."
If you need any help with how documentation works or anything, please let me know in telegram group :)

Reply:
That is being checked for each function inside it, so I thought not to add it.

Reply:
@LiraLemur Okay, I will create a new PR for documentation.:)
If I am not wrong these are the steps to be followed for documentation ... https://github.com/hyperledger/iroha/tree/main/docs

Reply:
>  If I am not wrong these are the steps to be followed for documentation ... https://github.com/hyperledger/iroha/tree/main/docs

Yes, You should:

1. checkout iroha
2. inside the main iroha directory: `cd docs/source`
3. `pip install -r requirements.txt`
4. `make html`
5. "After that, the documentation is generated in html format in _build folder"

Reply:
@ayush-jalan you need to change the source files in docs/source - those are mostly .rst files. Please contact me in telegram if you have questions about that format of general questions on readthedocs documentation.